addons/mail/static/src/core/thread_service.js

```diff
@@ -139,46 +163,76 @@ export class ThreadService {
         await this.orm.silent.call("mail.channel", "channel_fetched", [[thread.id]]);
     }
 
+    async fetchPinnedMessages(thread) {
+        if (
+            thread.model !== "mail.channel" ||
+            ["loaded", "loading"].includes(thread.pinLoadState)
+        ) {
+            return;
+        }
+        thread.pinLoadState = "loading";
+        try {
+            const messages = await this.rpc("/mail/channel/pins", {
```

Comment:
Name is unclear, better have it `/mail/channel/pinned_messages` or `/mail/channel/messages/pinned`